src/AspNetCoreModuleV2/InProcessRequestHandler/dllmain.cxx

```diff
@@ -103,31 +111,47 @@ __stdcall
 CreateApplication(
     _In_  IHttpServer        *pServer,
     _In_  IHttpContext       *pHttpContext,
-    _In_  PCWSTR             pwzExeLocation,
+    _In_  PCWSTR              pwzExeLocation,
     _Out_ IAPPLICATION      **ppApplication
 )
 {
-    HRESULT      hr = S_OK;
-    IAPPLICATION *pApplication = NULL;
-    REQUESTHANDLER_CONFIG *pConfig;
+    HRESULT                 hr = S_OK;
+    IN_PROCESS_APPLICATION *pApplication = NULL;
+    REQUESTHANDLER_CONFIG  *pConfig;
+
     // Initialze some global variables here
     InitializeGlobalConfiguration(pServer);
 
     hr = REQUESTHANDLER_CONFIG::CreateRequestHandlerConfig(pServer, pHttpContext->GetApplication(), &pConfig);
     if (FAILED(hr))
     {
-        return hr;
+        goto Finished;
     }
 
-    pApplication = new IN_PROCESS_APPLICATION(pServer, pConfig, pwzExeLocation);
-    if (pApplication == NULL)
+    pApplication = new IN_PROCESS_APPLICATION(pServer, pConfig);
+
+    hr = pApplication->Initialize(pwzExeLocation);
```

Comment:
if Initialize failed, you will double delete pConfig which will caused undefined behavior. first is done by the destructor of in_process_application and then by the function. This is why we usually use reference counter

Reply:
or you can simply set pConfig to NULL after new success indicating this config object belongs to application